src/ext/java/org/opentripplanner/ext/vectortiles/layers/vehicleparkings/VehicleParkingGroupsLayerBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.ext.vectortiles.layers.vehicleparkings;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
+import org.opentripplanner.ext.vectortiles.LayerBuilder;
+import org.opentripplanner.ext.vectortiles.PropertyMapper;
+import org.opentripplanner.ext.vectortiles.VectorTilesResource;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingService;
+import org.opentripplanner.transit.service.TransitService;
+import org.opentripplanner.util.geometry.GeometryUtils;
+
+public class VehicleParkingGroupsLayerBuilder extends LayerBuilder<VehicleParkingAndGroup> {
+
+  static Map<VehicleParkingGroupsLayerBuilder.MapperType, Function<Graph, PropertyMapper<VehicleParkingAndGroup>>> mappers = Map.of(
+    VehicleParkingGroupsLayerBuilder.MapperType.Digitransit,
+    g -> DigitransitVehicleParkingGroupPropertyMapper.create()
+  );
+  private final Graph graph;
+
+  public VehicleParkingGroupsLayerBuilder(
+    Graph graph,
+    TransitService transitService,
+    VectorTilesResource.LayerParameters layerParameters
+  ) {
+    super(
+      layerParameters.name(),
+      mappers
+        .get(VehicleParkingGroupsLayerBuilder.MapperType.valueOf(layerParameters.mapper()))
+        .apply(graph)
+    );
+    this.graph = graph;
+  }
+
+  @Override
+  protected List<Geometry> getGeometries(Envelope query) {
+    VehicleParkingService service = graph.getVehicleParkingService();
+    if (service == null) {
+      return List.of();
+    }
+    return service
+      .getVehicleParkingGroups()
+      .asMap()
```

Comment:
I don't know if using Multimap is actually a good idea here as I kinda have to convert it into a map here which is the only place where the vehicle parking groups are currently used.

Reply:
Since you're the only user, I think you can decide that.

Reply:
asMap is basically just exposing the internal map, so it is very cheap operation